js/classes/entities/midground/enemies/ground_enemy_base.js

```diff
@@ -92,6 +93,14 @@ class GroundEnemyBase {
         }
     }
 
+    knockback(amount) {
+        if (this.getDirection() == 1) {
+            this.enemy.pos.x -= amount;
```

Comment:
Generally, we like to avoid directly mutating a `Vector`, and to me this use case is not an exception. I would create a new vector, apply `amount` to it, and then assign it to `this.enemy.pos`.

Reply:
Good observation. I was in a rush at the time and completely forgot Vectors weren't supposed to be mutated.

Reply:
All fixed :)

Reply:
Awesome, thank you!